src/smolagents/local_python_executor.py

```diff
@@ -116,15 +116,24 @@ def custom_print(*args):
 
 DANGEROUS_MODULES = [
     "builtins",
+    "code",
+    "codecs",
+    "distutils",
+    "glob",
     "io",
     "multiprocessing",
     "os",
     "pathlib",
+    "pickle",
+    "posix",
     "pty",
+    "shelve",
```

Comment:
```suggestion
    "runpy",
    "shelve",
```
I would add that as well